posts/apache-spark-3.0.md

```diff
@@ -0,0 +1,54 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- Apache Spark
+- Big Data
+date: 2020-06-23T14:50:55.000Z
+title: "Apache Spark 3.0"
+image:
+---
+
+Databricks recently announced the release of [Apache Spark 3.0](https://databricks.com/blog/2020/06/18/introducing-apache-spark-3-0-now-available-in-databricks-runtime-7-0.html) with their Databricks Runtime 7.0. It's been a long road since the initial 1.0 release in 2014 which makes Spark one of the most active open-source big data projects with over 1000 contributors! This giant open-source community (one of the most active from the Apache Software Foundation) resolved over 3400 Jira tickets which is a pretty good way to celebrate its 10 years anniversary as an open-source project.
+
+# New features introduced
+
+- Java 11 support
+- Hadoop 3 support
+- Better ANSI SQL compatibility
+- 2x performance improvement on TPC-DS over Spark 2.4
+- Simpler/better PySpark exceptions
+- Up to 40x speedups for calling R user-defined functions
+- Structured Streaming UI
+
+Check the [migration guide](https://spark.apache.org/docs/3.0.0/migration-guide.html) in order to switch your code to the new version and also note that no major code changes should be required.
+
+# Python API improvements
+
+I've been always a fan of Python for its simplicity, libraries and capacity of doing more with less code. [PySpark](https://spark.apache.org/docs/latest/api/python/index.html) is great, especially if you want to use ML or NLP libraries, but the overall performances are always gonna be better on a Scala environment.
+
+Recently, Python passed Scala on the most used language on Spark which is not a surprise since PySpark has more than 5 million monthly downloads on PyPI. The goal of this release was to improve the functionalities and usability of PySpark. That includes a redesign of the Pandas UDF API using Python type hints, new Pandas UDF types, and more Pythonic error handling. Pandas is still my most favorite library for data analysis/manipulation but it is a single-node processing library which doesn't really fit in a distributed environment. I recommend using [Koalas](https://koalas.readthedocs.io/en/latest/) which is basically the Pandas API on Apache Spark. The Koalas API gives a coverage close to 80% of the Pandas API which is enough for most cases.
+
+PySpark errors are not always user-friendly to Python users because Spark was originally designed to run on a JVM. This release will simplify PySpark exceptions by making them more Pythonic and will also avoid showing unnecessary JVM stack trace.
```

Comment:
```suggestion
PySpark errors are not always user-friendly to Python users because Spark was originally designed to run on a JVM. This release will simplify PySpark exceptions by making them more Pythonic and will also avoid showing unnecessary JVM stack traces.
```